### PR TITLE
Add Ubuntu 24.04 LTS for Azure

### PR DESCRIPTION
**Pull request by mboersma**
## Change description

Adds support for Ubuntu 24.04-based builds on Azure.

## Related issues

- Refs #1406
- See #961 for prior art

### CODE REVIEW

**Comment by mboersma** [comment]
The two targets `build-azure-sig-ubuntu-2404-cvm` and `build-azure-sig-ubuntu-2404-gen2` are failing:

```shell
% PACKER_LOG=1 noproxy=* make -C images/capi build-azure-sig-ubuntu-2404-gen2
hack/ensure-python.sh
Checking if python is available
Python 3.12.3
...
2024/04/26 10:07:34 On error: 

A build command cannot run without at least one build to process. If the only or
except flags have been specified at run time check that at least one build is
selected for execution.

Error: No builds to run

A build command cannot run without at least one build to process. If the only or
2024/04/26 10:07:34 [INFO] (telemetry) Finalizing.
except flags have been specified at run time check that at least one build is
selected for execution.


2024/04/26 10:07:34 waiting for all plugin processes to complete...
make: *** [build-azure-sig-ubuntu-2404-gen2] Error 1
```

I'm having trouble figuring out where this is going wrong. The other new targets all work. Any help would be appreciated!

**Comment by mboersma** [comment]
Still getting `Error: No builds to run` on `make build-azure-sig-ubuntu-2404-cvm` and `make build-azure-sig-ubuntu-2404-gen2`. I'm failing to figure out why--any clues would be much appreciated!

File: images/capi/azure_targets.sh [hunk 1]
```diff
@@ -1,8 +1,8 @@
-VHD_TARGETS="ubuntu-2004 ubuntu-2204 centos-7 mariner-2 rhel-8 windows-2019-containerd windows-2022-containerd"
-VHD_CI_TARGETS="ubuntu-2004 ubuntu-2204 mariner-2 windows-2019-containerd windows-2022-containerd"
-SIG_TARGETS="ubuntu-2004 ubuntu-2204 centos-7 mariner-2 rhel-8 windows-2019-containerd windows-2022-containerd flatcar"
-SIG_CI_TARGETS="ubuntu-2004 ubuntu-2204 mariner-2 windows-2019-containerd windows-2022-containerd flatcar"
-SIG_GEN2_TARGETS="ubuntu-2004 ubuntu-2204 centos-7 mariner-2 flatcar"
-SIG_GEN2_CI_TARGETS="ubuntu-2004 ubuntu-2204 mariner-2 flatcar"
-SIG_CVM_TARGETS="ubuntu-2004 ubuntu-2204 windows-2019-containerd windows-2022-containerd"
-SIG_CVM_CI_TARGETS="ubuntu-2204 windows-2022-containerd"
+VHD_TARGETS="ubuntu-2004 ubuntu-2204 ubuntu-2404 centos-7 mariner-2 rhel-8 windows-2019-containerd windows-2022-containerd"
+VHD_CI_TARGETS="ubuntu-2204 ubuntu-2404 mariner-2 windows-2019-containerd windows-2022-containerd"
```
Comment (mboersma) [inline images/capi/azure_targets.sh:2]: I traded Ubuntu 20.04 for 24.04 in CI, not wanting to expand our costs per PR run. But please let me know if this doesn't seem like a worthwhile tradeoff.

File: images/capi/Makefile [hunk 2]
```diff
@@ -488,7 +488,7 @@ $(AZURE_BUILD_SIG_GEN2_TARGETS): deps-azure
 
 .PHONY: $(AZURE_BUILD_SIG_CVM_TARGETS)
 $(AZURE_BUILD_SIG_CVM_TARGETS): deps-azure
-	. $(abspath packer/azure/scripts/init-sig.sh) $(subst build-azure-sig-,,$@) && packer build $(if $(findstring windows,$@),$(PACKER_WINDOWS_NODE_FLAGS),$(PACKER_NODE_FLAGS)) -var-file="$(abspath packer/azure/azure-config.json)" -var-file="$(abspath packer/azure/azure-sig-cvm.json)" -var-file="$(abspath packer/azure/$(subst build-azure-sig-,,$@).json)" -only="$(subst build-azure-,,$@)" $(ABSOLUTE_PACKER_VAR_FILES) packer/azure/packer$(findstring -windows,$@).json
+	. $(abspath packer/azure/scripts/init-sig.sh) $(subst build-azure-sig-,,$@) && $(PACKER) build $(if $(findstring windows,$@),$(PACKER_WINDOWS_NODE_FLAGS),$(PACKER_NODE_FLAGS)) -var-file="$(abspath packer/azure/azure-config.json)" -var-file="$(abspath packer/azure/azure-sig-cvm.json)" -var-file="$(abspath packer/azure/$(subst build-azure-sig-,,$@).json)" -only="$(subst build-azure-,,$@)" $(ABSOLUTE_PACKER_VAR_FILES) packer/azure/packer$(findstring -windows,$@).json
```
Comment (AverageMarcus) [inline images/capi/Makefile:491]: Nice catch! 👍

**Review by AverageMarcus — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**